packages/scheduler/lib/workers/scheduling/scheduling.worker.ts

```diff
@@ -95,49 +96,68 @@ export class SchedulingChild {
     async schedule(): Promise<void> {
         const res = await this.db.transaction(async (trx): Promise<Result<string[]>> => {
             const taskIds: string[] = [];
-            // Try to acquire a lock to prevent multiple instances from scheduling at the same time
-            const res = await trx.raw('SELECT pg_try_advisory_xact_lock(?) AS lock_granted', [5003001106]);
-            const lockGranted = res?.rows.length > 0 ? res.rows[0].lock_granted : false;
-
-            if (lockGranted) {
-                const schedules = await dueSchedules(trx);
-                if (schedules.isErr()) {
-                    return Err(`Failed to get due schedules: ${stringifyError(schedules.error)}`);
-                } else {
-                    const createTasks = schedules.value.map((schedule) =>
-                        tasks.create(trx, {
-                            scheduleId: schedule.id,
-                            startsAfter: new Date(),
-                            name: `${schedule.name}:${new Date().toISOString()}`,
-                            payload: schedule.payload,
-                            groupKey: schedule.groupKey,
-                            retryCount: 0,
-                            retryMax: schedule.retryMax,
-                            createdToStartedTimeoutSecs: schedule.createdToStartedTimeoutSecs,
-                            startedToCompletedTimeoutSecs: schedule.startedToCompletedTimeoutSecs,
-                            heartbeatTimeoutSecs: schedule.heartbeatTimeoutSecs
-                        })
-                    );
-                    const res = await Promise.allSettled(createTasks);
-                    for (const taskRes of res) {
-                        if (taskRes.status === 'rejected') {
-                            logger.error(`Failed to schedule task: ${taskRes.reason}`);
-                        } else if (taskRes.value.isErr()) {
-                            logger.error(`Failed to schedule task: ${stringifyError(taskRes.value.error)}`);
-                        } else {
-                            taskIds.push(taskRes.value.value.id);
+            const span = tracer.startSpan('scheduler.scheduling.schedule');
```

Comment:
note if you don't wrap in a callback the next calls will not be wrapped in this span

Reply:
fixed in https://github.com/NangoHQ/nango/pull/2374/commits/a5d1699e88a4933329438cf3abc5160ac36e6e80


Looks nicer now !
[Screenshot 2024-06-20 at 09 53 54](https://github.com/NangoHQ/nango/assets/233326/fb0c534b-eb3f-457d-95b8-3d1dd2affb60)